### ISSUE TITLE
odroid xu4 on odroid-5.16.y and up

### ISSUE THREAD

**Issue by paralin** [tobetter/linux#35]
Hi @tobetter 

Thank you for your work on this as always,

I noticed that in odroid-5.15.y there is odroidxu4_defconfig, but not in odroid-5.16.y

Where did odroidxu4_defconfig go, and which defconfig should I use instead now?

Thanks :)

**Comment by paralin** [comment]
... and if there is a minimal defconfig to apply on top of the arch-specific defaults which works, I'd be happy to use that instead :)

**Comment by tobetter** [comment]
Well...actually I am about to give up to manage XU4 build since no one is using the kernel... :)

**Comment by paralin** [comment]
@tobetter you're not supporting xu4 anymore? I have several HC2s and a couple XU4s, definitely it's still used.

**Comment by paralin** [comment]
@tobetter Any news on this? What's your recommendation for what I should switch Skiff to in terms of supporting our existing XU4 users? I know of several active projects that are currently using XU4 and your 5.15.x branch and need to continue to receive updates going forward. I guess mainline sort-of works?

**Comment by paralin** [comment]
That said - I don't worry about managing the build, just about keeping a kernel tree up to date for it. If you're not interested in supporting it anymore, I guess best thing to do would be for me to start doing a branch and rebasing on newer kernel releases independently?

**Comment by paralin** [comment]
https://github.com/skiffos/linux/tree/skiff-odroid-5.16.y/

OK - I've re-based the xu4-specific commits on odroid-5.16.y and will continue to keep that up-to-date for now.

**Comment by uDude** [comment]
@paralin Many Thanks.  I still have many xu4 (exynos5's: maybe 35 or so) as well as older Exynos4 in use.  I wish Samsung had continued.  Their chips are one of very few to compete with Apple. :)  I use mainline, mostly, will happily switch if you are planning to keep it alive there. South Korean engineering is pretty awesome.

**Comment by paralin** [comment]
@uDude Wow, that's a lot of xu4s! I've got about 8 running. I've tested my branch & it works with 5.16.14. :)

**Comment by DylanVanAssche** [comment]
I use this kernel for 2 ODROID HC2s as well, I was surprised that it dropped support for them in 5.16 and higher...
I guess I will have to look into going fully mainline, most commits on top of mainline are related to GPU, HDMI, etc. only a few for thermal etc. Maybe trying to upstream them would help.

**Comment by paralin** [comment]
@DylanVanAssche the above kernel linked is confirmed to work on xu4, hc2, and n2. It's updated to latest 5.16

**Comment by DylanVanAssche** [comment]
@paralin Yes! That's awesome!

I went through the commits to see which ones are interesting for the HC2s I have here, this is what I found:


- e6b9c06fd20432c0b35e55902d8f6d7d0aa30234 ODROID-XU4: arm/dts: remove write-protect pin from SD card
- 489a28a85301ddc0b0da4885e9da4f70bb3a41d2 ODROID-XU4: arm/exynos: No need to use enynos_init_late
- e96f6b20411e0985d35e872862f37e6e56cca1f7 ODROID-XU4: devfreq: exynos-bus: workaround dev_pm_opp_set_rate() err
- 3aa79f3a1c1d46177b545d65e6bff3e383e73d92 ODROID-XU4: arm: dts: exynos5422: enable Exynos5422 TMU
- 38cb009cd3574c459c603a71d544adb03dea3a3d ODROID-XU4: regulator: s2mps11: add ethernet power reset in shutdown
- ee364448ec75704702db6fb80b605c8f75cbf496 ODROID-XU4: thermal: exynos: add support for 8 trip points on Exynos5
- b3c795564a8a1b289bb10a832e7b34cfa8d1a140 ODROID-XU4: Update hack avoiding the invalid temperature by TMU broken
- ff0a5163feddf76aed98b03e4fbf843ee8aa07b5 ODROID-XU4: ARM: exynos: add machine description for ODROID-XU3/4
- d1cc5b6a5db863e1518608443bfa947046f984c4 ODROID-XU4: regulator: s2mps11: call shutdown function to poweroff

(skipped the defconfig commits as I have a config myself in postmarketOS and still have to see which are are in an upstreamable state)

Submitting those to the mainline kernel would be great to have 100% mainline running on the HC2.
I maintain this device in postmarketOS and moving to the mainline kernel would be so much better for maintenance :)
I heavily depends on my HC2s as all my infrastructure runs on it.

**Comment by paralin** [comment]
@DylanVanAssche I agree upstreaming would be great, but I don't think I have the relevant knowledge of what those commits are modifying to do that job.
